dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java

```diff
@@ -1609,16 +1611,19 @@ public ContentletHashMap() {
 			super();
 		}
 
-		public Object put(final String key, final Object value) {
+		public Object put(final String key, final Object newValue) {
 
-			Contentlet.this.markAsDirty();
+		    final Object oldValue = this.get(key);
```

Comment:
what happen if oldValue is null?

Reply:
`equals` will return `false` and the contentlet will be marked as dirty if  `newValue != null`: 

```
public static boolean equals(Object a, Object b) {
        return (a == b) || (a != null && a.equals(b));
    }
```

Otherwise, the key will be removed.